stackit/internal/services/iaas/networkarearegion/resource.go

```diff
@@ -0,0 +1,697 @@
+										validate.UUID(),
+										validate.NoSeparator(),
+									},
+								},
+								"prefix": schema.StringAttribute{
+									Description: "Classless Inter-Domain Routing (CIDR).",
+									Required:    true,
+								},
+							},
+						},
+					},
+					"transfer_network": schema.StringAttribute{
+						Description: "IPv4 Classless Inter-Domain Routing (CIDR).",
+						Required:    true,
+						PlanModifiers: []planmodifier.String{
+							stringplanmodifier.RequiresReplace(),
+						},
+					},
+					"default_prefix_length": schema.Int64Attribute{
+						Description: "The default prefix length for networks in the network area.",
+						Optional:    true,
+						Computed:    true,
+						Validators: []validator.Int64{
+							int64validator.AtLeast(24),
+							int64validator.AtMost(29),
+						},
+						Default: int64default.StaticInt64(25),
+					},
+					"max_prefix_length": schema.Int64Attribute{
+						Description: "The maximal prefix length for networks in the network area.",
+						Optional:    true,
+						Computed:    true,
+						Validators: []validator.Int64{
+							int64validator.AtLeast(24),
+							int64validator.AtMost(29),
+						},
+						Default: int64default.StaticInt64(29),
+					},
+					"min_prefix_length": schema.Int64Attribute{
+						Description: "The minimal prefix length for networks in the network area.",
+						Optional:    true,
+						Computed:    true,
+						Validators: []validator.Int64{
+							int64validator.AtLeast(8),
+							int64validator.AtMost(29),
+						},
+						Default: int64default.StaticInt64(24),
+					},
+				},
+			},
+		},
+	}
+}
+
+// Create creates the resource and sets the initial Terraform state.
+func (r *networkAreaRegionResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) { // nolint:gocritic // function signature required by Terraform
+	// Retrieve values from plan
+	var model Model
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &model)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	organizationId := model.OrganizationId.ValueString()
+	networkAreaId := model.NetworkAreaId.ValueString()
+	region := r.providerData.GetRegionWithOverride(model.Region)
+	ctx = tflog.SetField(ctx, "organization_id", organizationId)
+	ctx = tflog.SetField(ctx, "network_area_id", networkAreaId)
+	ctx = tflog.SetField(ctx, "region", region)
+
+	// Generate API request body from model
+	payload, err := toCreatePayload(ctx, &model)
+	if err != nil {
+		core.LogAndAddError(ctx, &resp.Diagnostics, "Error creating network area region", fmt.Sprintf("Creating API payload: %v", err))
+		return
+	}
+
+	// Create new network area region configuration
+	keyPair, err := r.client.CreateNetworkAreaRegion(ctx, organizationId, networkAreaId, region).CreateNetworkAreaRegionPayload(*payload).Execute()
+	if err != nil {
+		core.LogAndAddError(ctx, &resp.Diagnostics, "Error creating network area region", fmt.Sprintf("Calling API: %v", err))
+		return
+	}
+
```

Comment:
Log response / trace id
```suggestion
	ctx = core.LogResponse(ctx)

```